### PR TITLE
#80 Returned event after creating; Added organizer name to EventDto

### PR DESCRIPTION
**Pull request by adrianczekaj**

### CODE REVIEW

File: GiftGivingGenerator.API/Controllers/EventsController.cs [hunk 1]
```diff
@@ -43,8 +43,9 @@ public ActionResult CreateEvent([FromBody] CreateEventWithPersonsDto dto)
 		}
 
 		var eventId = _eventRepository.Insert(@event);
+		var eventToReturn = _eventRepository.Get<EventDto>(eventId);
 		
-		return CreatedAtAction(nameof(GetEventWithPersonsAndExclusions), new {id = @eventId}, null);
+		return Ok(eventToReturn);
```
Comment (maciejczekaj) [inline GiftGivingGenerator.API/Controllers/EventsController.cs]: Why this is changed? You can return object as 3 parameter of CreatedAtAction

**Review by maciejczekaj — CHANGES_REQUESTED** [review]

File: GiftGivingGenerator.API/Controllers/EventsController.cs (hunk 1 above)
Comment (adrianczekaj) [reply GiftGivingGenerator.API/Controllers/EventsController.cs]: In this conversation is missing Paulina :)
In the first solution, we return information in HEAD of response, in the second - in BODY of response.
Paulina said that she didn't know how to get information from HEAD and she always gets it from BODY.

Comment (maciejczekaj) [reply GiftGivingGenerator.API/Controllers/EventsController.cs]: But you can do both. Return response with code `201` and with response body:
- https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.controllerbase.createdataction?view=aspnetcore-6.0
- https://stackoverflow.com/questions/1860645/create-request-with-post-which-response-codes-200-or-201-and-content

Comment (adrianczekaj) [reply GiftGivingGenerator.API/Controllers/EventsController.cs]: Ok, still is no 100% clear.

There are 2 main possibilities:

1. `Ok()` - returns created entity **without** reference to it.
2. `CreatedAtAction(nameof(CreateEvent), new {id = eventId}, eventWithOrganizerDto)` - returns created entity **with** reference to it.

The question is do we need solution 2 with reference if we know that Paulina won't use it?

By the way, I have a second question.
After using `Ok(eventDto)` I realized that maybe I shouldn't return the object I just created as dto.
Should I use `Ok(justCreatedEntity)` (without dto) always when I'm creating an entity?

Comment (maciejczekaj) [reply GiftGivingGenerator.API/Controllers/EventsController.cs]: 1. Paulina may not be the only consumer of this API so yes, we need a reference to this resource.
2. You shouldn't return entity (always return dto)

Comment (adrianczekaj) [reply GiftGivingGenerator.API/Controllers/EventsController.cs]: Thanks a lot - I fixed it.

**Review by maciejczekaj — APPROVED** [review]

**Pull request merged**